src/plugins/auto_batch/tests/functional/behavior/ov_executable_network/properties.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "behavior/ov_executable_network/properties.hpp"
+
+#include "ie_system_conf.h"
+#include "openvino/runtime/properties.hpp"
+
+using namespace ov::test::behavior;
+
+namespace {
+
+const std::vector<ov::AnyMap> auto_batch_inproperties = {
+    {ov::num_streams(-100)},
+    {{CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG), std::string(CommonTestUtils::DEVICE_CPU) + "(4)"},
+     {ov::auto_batch_timeout(-1)}},
+#ifdef HAVE_TEMPLATE_PLUGIN
+    {ov::device::id("UNSUPPORTED_DEVICE_ID_STRING")},
+    {{CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG), std::string(CommonTestUtils::DEVICE_TEMPLATE) + "(4)"},
+     {ov::auto_batch_timeout(-1)}},
+#endif
+};
```

Comment:
why do we have template plugin under conditional compilation, while CPU used without any guards?
As I wrote, better not rely on any plugin at all and use mocks.

Reply:
For AUTO_BATCH plugin tests, any hw plugin shouldn't be involved in, but we can leverage mock plugin to help create test case, in some special test cases we can leverage TEMPLATE plugin if possible.